src/Compilers/CSharp/Portable/Symbols/BaseTypeAnalysis.cs

```diff
@@ -288,6 +299,7 @@ private static (ThreeState isManaged, bool hasGenerics) IsManagedTypeHelper(Name
                 case TypeKind.Enum:
                     return (ThreeState.False, hasGenerics);
                 case TypeKind.Struct:
+                case TypeKind.Extension:
```

Comment:
>case TypeKind.Extension: [](http://example.com/codeflow?start=16&length=24)

Can extension have any instance fields?

Reply:
Extensions disallow instance fields

Reply:
> Extensions disallow instance fields

Then the result is known. Only presence of fields can have any effect

Reply:
I had `IsManaged` return true in error scenarios where a field is declared in an extension. Sounds like you think we should just return false always.